source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
 
-- Description
-  - Define term: discriminator
-- Example
-  - Show each way (annotation and ClassModelBuilder method) to enable discriminator
-  
+.. code-block:: java
+   :emphasize-lines: 4
 
-Supporting POJOs without Default Constructors
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   public class CourteousAgeSerialization implements PropertySerialization<Integer> {
+       @Override
+       public boolean shouldSerialize(Integer value) {
+           return (value < 30);
+       }
+   }
 
-- Description, explaining when this is used
-- Example of BsonCreator and BsonProperty annotations to specify the constructor and parameter names to use
+The class above specifies that any integer above 29 is not serialized and
+therefore, not included in the MongoDB document. Suppose you applied this
+custom serialization behavior to the following sample POJO:
 
-Customizing Serialization
-~~~~~~~~~~~~~~~~~~~~~~~~~
+.. code-block:: java
 
-- Description, when to use
-- Examples
-  - Annotation
-  - PropertySerialization in PropertyModelBuilder
+   public class BirthdayInvitation {
+       private String name;
+       private Integer age;
+       private LocalDateTime eventDateTime;
 
+       // ...
+   }
+
+You can specify the custom serialization by adding the
+``CourteousAgeSerialization`` instance to the ``PropertyModelBuilder`` from
+the ``ClassModel`` property associated with the ``age`` field using the
+following code:
+
+.. code-block:: java
+
+   ClassModelBuilder<BirthdayInvitation> classModel = ClassModel.builder(BirthdayInvitation.class);
+   ((PropertyModelBuilder<Integer>) classModel.getProperty("age"))
+           .propertySerialization(new CourteousAgeSerialization());
+
+   PojoCodecProvider pojoCodecProvider = PojoCodecProvider.builder().register(classModel.build()).build();
+   CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+If you insert a POJO that contains a value greater than 29 in the ``age``
+field, the serialized document omits it. The resulting document should
+resemble the following:
+
+.. code-block:: json
+   :copyable: false
+
+   { "_id" : ObjectId("..."), "eventDateTime" : ..., "name" : "Galadriel" }
 
 Generics Support
 ^^^^^^^^^^^^^^^^
 
-- Description of when a POJO is not serializable
-- Example showing how to use PropertyCodecProviders to provide concrete class information
+You can use the POJO ``Codec`` to serialize classes that contain generic
+properties if they meet the following criteria:
+
+- contain only bounded concrete type parameters
+- the top-level POJO does not contain any type parameters
+
+The ``ClassModelBuilder`` inspects and saves concrete type parameters to work
+around type erasure. It cannot serialize classes that contain generic
+properties without concrete type parameters since the JVM removes the type
+parameter information.
+
+Instead, you can implement the ``PropertyCodecProvider`` interface to specify
```

Comment:
Instead of using the default ClassModelBuilder? I would consider ditching the "Instead" here or maybe elaborate more about the options before we descend into the details in these two paragraphs.